### PR TITLE
Models specs and validations

### PR DESCRIPTION
**Pull request by Ochiengsteven**
### Changes Made

1. Added validations to enhance data integrity for the User and Post models. 🛡️
2. Implemented unit specs for all models' methods and validations. 🧪

#### User Model

1. `Name` must not be blank.
2. `PostsCounter` must be an integer greater than or equal to zero.

#### Post Model

1. `Title` must not be blank.
2. `Title` must not exceed 250 characters.
3. `CommentsCounter` must be an integer greater than or equal to zero.
4. `LikesCounter` must be an integer greater than or equal to zero.

### Unit Specs

1. Implemented unit tests for all models' methods and validations.
2. Ensured that validations are working as expected.

> - Validations added to prevent invalid data entry and ensure data consistency.
> - Unit specs provide comprehensive coverage for models' behavior.

### Testing

1. Ran unit specs to validate model methods and ensure they are error-free.
2. Manually tested validations in the 'rails console' to verify their functionality.

### completed

- [x] Validations added for User and Post models.
- [x] Unit specs implemented for all models' methods and validations.
- [x] README updated with relevant information.

Please review and allow merge.

Thank you! 🙌

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  it 'should be valid when likes_counter is zero' do
+    subject.likes_counter = 0
+    expect(subject).to be_valid
+  end
+
+  it 'should be valid when likes_counter is positive' do
+    subject.likes_counter = 5
+    expect(subject).to be_valid
+  end
```
Comment (TedLivist) [inline spec/models/post_spec.rb]: - [ ] The custom methods in the models should be tested. Let's add tests for the `update_author_posts_counter` and `recent_comments` methods in the Post model.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  it 'should be valid when posts_counter is zero' do
+    subject.posts_counter = 0
+    expect(subject).to be_valid
+  end
+
+  it 'should be valid when posts_counter is positive' do
+    subject.posts_counter = 5
+    expect(subject).to be_valid
+  end
+end
```
Comment (TedLivist) [inline spec/models/user_spec.rb]: - [ ] Kindly add tests for the `recent_posts` method in the User model.

- [ ] You are missing tests for the Comment and Like models. Kindly add them and ensure the custom methods there are also well tested.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Ochiengsteven,

Good job so far!
✔️ Well-written and passing tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 2 above)
Comment (julie-ify) [inline spec/models/user_spec.rb]: - [ ] As suggested by the previous reviewer, kindly add test for the recent posts method in the user model file.

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @Ochiengsteven, 

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 2 above)
Comment (Ochiengsteven) [reply spec/models/user_spec.rb]: @julie-ify Thanks for the feedback, I've already added the test for the method and resubmitted for review

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Ochiengsteven👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

image

## To Highlight 🎉
- All tests are passed. ✔️ 
- Good job with validations. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**